servers/src/main/java/tachyon/master/lineage/meta/LineageStore.java

```diff
@@ -186,6 +160,21 @@ public synchronized Lineage getLineage(long lineageId) {
   }
 
   /**
+   * Gets the lineage that has the given output file.
+   *
+   * @param fileId the file id
+   * @return the lineage containing the output file
+   * @throws LineageDoesNotExistException
+   */
+  public synchronized Lineage getLineageOfOutputFile(long fileId)
+      throws LineageDoesNotExistException {
+    Lineage lineage = mOutputFileIndex.get(fileId);
+    LineageDoesNotExistException.check(lineage != null, ExceptionMessage.LINEAGE_DOES_NOT_EXIST,
```

Comment:
I only see readability as the proposed advantage, but I think this decreases readability unless the user already knows what `LineageDoesNotExistException.check` does. It also looks superfluous to have both `LineageDoesNotExistException` and `ExceptionMessage.LINEAGE_DOES_NOT_EXIST`. This is more of a nit though - lets discuss it another time outside of this PR.